File: leetcode/AIcoding/3.DeepSeek-v3/go_solutions/2928-distribute-candies-among-children-i.go
func distributeCandies(n int, limit int) int64 {
    var count int64 = 0
    for a := 0; a <= limit; a++ {
        for b := 0; b <= limit; b++ {
            c := n - a - b
            if c >= 0 && c <= limit {
                count++
            }
        }
    }
    return count
}

File: leetcode/AIcoding/2.Gemini-2.0/c_solutions/2928-distribute-candies-among-children-i.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int distributeCandies(int n, int limit) {
    int count = 0;
    for (int i = 0; i <= limit; i++) {
        if (n - i >= 0 && n - i <= 2 * limit) {
            int remaining = n - i;
            int min_val = (remaining > limit) ? remaining - limit : 0;
            count += limit - min_val + 1;
        }
    }
    return count;
}